crates/next-core/js/src/internal/chunk-loader.ts

```diff
@@ -0,0 +1,84 @@
+// Adapted from https://github.com/vercel/next.js/blob/canary/packages/next/client/route-loader.ts
+
+import { markAssetError } from "next/dist/client/route-loader";
+
+const loadedScripts: Map<string, Promise<unknown>> = new Map();
+const styleSheets: Map<string, Promise<HTMLLinkElement>> = new Map();
+
+export function maybeExecuteScript(src: string): Promise<unknown> {
+  let prom = loadedScripts.get(src.toString());
+  if (prom) {
+    return prom;
+  }
+
+  // Skip executing script if it's already in the DOM:
+  if (document.querySelector(`script[src^="${src}"]`)) {
+    return Promise.resolve();
+  }
+
+  loadedScripts.set(src.toString(), (prom = appendScript(src)));
```

Comment:
Can we please avoid assignment expression in JS 🙏 They are rarely necessary, and usually a source of confusion.

Reply:
It's copy pasted from next.js, but I can change it

Reply:
Ah, makes sense then 😅 If we've already made changes here, I'd say let's change this as well.